subject.py

```diff
@@ -0,0 +1,178 @@
+import datajoint as dj
+
+import reference
+
+schema = dj.schema(dj.config['names.%s' % __name__], locals())
+
+
+
+@schema
+class Species(dj.Manual):
+    # <class 'subjects.models.Species'>
+    definition = """
+    species_id:			int             # species id
+    ---
+    binomial:			varchar(255)	# binomial
+    display_name:		varchar(255)	# display name
+    """
+
+
+@schema
+class Strain(dj.Manual):
+    # <class 'subjects.models.Strain'>
+    definition = """
+    strain_id:			int             # strain id
+    ---
+    descriptive_name:		varchar(255)	# descriptive name
+    description:    		varchar(255)	# description
+    """
+
+
+@schema
+class Sequence(dj.Manual):
+    # <class 'subjects.models.Sequence'>
+    definition = """
+    sequence_id:		int             # sequence id
+    ---
+    base_pairs:			varchar(255)	# base pairs
+    description:		varchar(255)	# description
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Allele(dj.Manual):
+    # <class 'subjects.models.Allele'>
+    definition = """
+    allele_id:			int             # allele id
+    ---
+    standard_name:		varchar(255)	# standard name
+    informal_name:		varchar(255)	# informal name
+    """
+
+
+@schema
+class Line(dj.Manual):
+    # <class 'subjects.models.Line'>
+    # TODO: autoname index: basically autogenerates a unique sequence
+    definition = """
+    -> Species
+    -> Strain
+    -> Sequence
+    line_id:				int             # line id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    target_phenotype:			varchar(255)	# target phenotype
+    auto_name:				varchar(255)	# auto name
+    is_active:				boolean		# is active
+    subject_autoname_index:		integer		# subject autoname index
+    breeding_pair_autoname_index:	integer		# breeding pair autoname index
+    litter_autoname_index:		integer		# litter autoname index
+    """
+
+
+@schema
+class Source(dj.Manual):
+    # <class 'subjects.models.Source'>
+    definition = """
+    source_id:				int             # source id
+    ---
+    name:				varchar(255)	# name
+    description:			varchar(255)	# description
+    """
+
+
+@schema
+class SubjectRequest(dj.Manual):
+    # <class 'subjects.models.SubjectRequest'>
+    definition = """
+    -> reference.User
+    -> Line
+    subject_request_id:			int     	# subject request id
+    ---
+    count:				integer		# count
+    date_time:				date		# date time
+    due_date:				date		# due date
+    description:			varchar(255)	# description
+    """
+
+
+@schema
+class Subject(dj.Manual):
```

Comment:
This appears to be too heavily populated with attributes. It'd make sense to break off some attributes into its own table, especially given it is unlikely that every attribute will be filled for every mouse (even if that's the case eventually, I'd imagine many mouse will not have i.e. implant for some period of time).

Also markers like `to_be_genotype` is not a good attribute as that'd expected to change. Rather, there should be a table like `GenoType` that points to `Subject` and one should determine if the mouse needs to be genotyped or not by checking `GenoType & Subject` presence.

Reply:
will zap the state tracking fields - as for attributes, any pointers on what to break down where would be welcome. seemed alot to me as well, but wasn't sure what/where to reorganize.

Reply:
Yep - let me look through a bit more in detail and come back with suggestion.